### PR TITLE
[Core] Choose most recent framework from available frameworks as default...

### PR DESCRIPTION
**Pull request by loganek**
....

This commit fixes bug #24458.

### CODE REVIEW

**Comment by monoadmin** [comment]
Hello! I'm the build bot for the Mono project. I need approval from a Mono team member to build this pull request. A team member should reply with "approve" to approve a build of this pull request, "whitelist" to whitelist this and all future pull requests from this contributor, or "build" to explicitly request a build, even if one has already been done. Contributors can ignore this message.

**Comment by mhutch** [comment]
Whitelist

File: main/src/core/MonoDevelop.Core/MonoDevelop.Projects/DotNetProject.cs [hunk 1]
```diff
@@ -359,9 +359,34 @@ public string StripImplicitNamespace (string ns)
...
 		}
-		
+
+		/// <summary>
+		/// Compares two frameworks based on id ascending, version descending, profile ascending.
+		/// </summary>
+		/// <returns>0 if both frameworks are equal, -1 if .</returns>
+		/// <param name="x">The x first framework target to compare.</param>
+		/// <param name="y">A 32-bit signed integer that indicates the relationship between the two target frameworks.</param>
+		public static int CompareFrameworks(TargetFramework x, TargetFramework y)
```
Comment (slluis) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Projects/DotNetProject.cs]: This should be done by implementing IComparable<TargetFramework> in TargetFramework.

File: main/src/core/MonoDevelop.Core/MonoDevelop.Projects/DotNetProject.cs [hunk 2]
```diff
@@ -359,9 +359,34 @@ public string StripImplicitNamespace (string ns)
 		/// </returns>
 		public virtual TargetFrameworkMoniker GetDefaultTargetFrameworkId ()
 		{
-			return Services.ProjectService.DefaultTargetFramework.Id;
+			var frameworks = Runtime.SystemAssemblyService.GetTargetFrameworks ()
```
Comment (slluis) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Projects/DotNetProject.cs]: It should not return a framework newer than Services.ProjectService.DefaultTargetFramework. Sometimes we add early support for frameworks which are not yet ready to be the default.

File: main/src/core/MonoDevelop.Core/MonoDevelop.Projects/DotNetProject.cs [hunk 3]
```diff
@@ -359,9 +359,34 @@ public string StripImplicitNamespace (string ns)
...
+				.Where (fx => !fx.Hidden && TargetRuntime.IsInstalled (fx) && SupportsFramework (fx))
+				.ToList ();
+			if (!frameworks.Any ())
+				throw new ArgumentException ("Compatible target framework does not exists.");
+
+			frameworks.Sort (CompareFrameworks);
```
Comment (slluis) [inline main/src/core/MonoDevelop.Core/MonoDevelop.Projects/DotNetProject.cs]: If TargetFramework implements IComparable, you can just use Linq Max

**Pull request closed without merging**